aesara/tensor/random/basic.py

```diff
@@ -431,6 +431,45 @@ def rng_fn_scipy(cls, rng, shape, scale, size):
 gamma = GammaRV()
 
 
+class StandardGammaRV(GammaRV):
```

Comment:
If we only intend to add `standard_gamma` for NumPy interface compatibility, then the `GammaRV` `Op` alone should suffice, no?  For instance, `standard_gamma` could be an `Op` constructor function that effectively removes the shape and rate arguments from `GammaRV`.

As always, we want to avoid adding new `Op`s whenever we reasonably can.

N.B. This is an example of the concern stated in https://github.com/aesara-devs/aemcmc/pull/67#discussion_r984062408.

Reply:
I think the issue is that you couldn't use RandomStream if it's not a specific Op. That's why the standard_nornalnwas made a subclass as well IIRC

Reply:
> If we only intend to add `standard_gamma` for NumPy interface compatibility, then the `GammaRV` `Op` alone should suffice, no?  For instance, `standard_gamma` could be an `Op` constructor function that effectively removes the shape and rate arguments from `GammaRV`.

Yes I completely agree with all this. We can probably monkey patch the existing RVs.

Only difficulty, as @ricardoV94 said, is `RandomStream` expects its attributes to be `RandomVariable`s.

Reply:
> If we only intend to add `standard_gamma` for NumPy interface compatibility, then the `GammaRV` `Op` alone should suffice, no? For instance, `standard_gamma` could be an `Op` constructor function that effectively removes the shape and rate arguments from `GammaRV`.
> 
> As always, we want to avoid adding new `Op`s whenever we reasonably can.
> 
> N.B. This is an example of the concern stated in [aesara-devs/aemcmc#67 (comment)](https://github.com/aesara-devs/aemcmc/pull/67#discussion_r984062408).

I agree with all this.

@ricardoV94 we can monkey patch the base classes. I'm not a big fan of this approach, but it should work:

```python
import aesara.tensor as at
from aesara.tensor.random.basic import NormalRV

def create_standard_normal():

    def new_call(size, **kwargs):
        return A.__call__(0, 1, size, **kwargs)

    C = NormalRV()
    C.__call__ = new_call
    return C

standard_normal = create_standard_normal()

print(type(normal))
# <class 'aesara.tensor.random.basic.NormalRV'>
print(type(standard_normal))
# <class 'aesara.tensor.random.basic.NormalRV'>

print(standard_normal().owner.inputs)
# [RandomGeneratorSharedVariable(<Generator(PCG64) at 0x7FEA4514E960>), TensorConstant{[]}, TensorConstant{11}, TensorConstant{0.0}, TensorConstant{1.0}]
```

Reply:
Does it work with RandomStream like that?

Reply:
I think it will, `RandomStream` checks that the attribute is both in `aesara.tensor.random.basic` and is an instance of `RandomVariable`. We'll soon know for sure.

Reply:
This quickly gets complicated, monkey patching affects the class globally so i would need to replace the method of the instance directly.

By the way, I've noticed that NumPy's `Generator` [has a a method for every RV](https://github.com/numpy/numpy/blob/main/numpy/random/_generator.pyx)

Reply:
Maybe we can add a canonical rewrite that replaces the standard versions immediately?

Or can we tweak RandomStream perhaps?

Reply:
> Maybe we can add a canonical rewrite that replaces the standard versions immediately?

That would certainly work. But I'd rather not rely on rewrites to "fix" a problem in the representation of objects in the IR, which is also my concern in https://github.com/aesara-devs/aesara/issues/1213. It would be better if they were represented by the same type in the original graph.

Reply:
Adding / Replacing most methods in class instances works fine, for instance:

```python
import aesara.tensor as at
from aesara.tensor.random.basic import t, StudentTRV

def create_standard_t():

    C = StudentTRV()

    def new_call(self, df, size=None, **kwargs):
        return self.__call__(df, 0, 1, size, **kwargs)

    C.call = new_call.__get__(C)
    return C

standard_t = create_standard_t()
print(standard_t.call(2.).owner.inputs)
# [RandomGeneratorSharedVariable(<Generator(PCG64) at 0x7F618591A420>), TensorConstant{[]}, TensorConstant{11}, TensorConstant{2.0}, TensorConstant{0}, TensorConstant{1}]
```
However, special methods like `__call__` are looked up with respect to the class of the object, and not its instance. So if we monkey patch the method it will end up affecting all the instances of the class, and thus instances of the non-standard `RandomVariable`. So I think that's a dead end.

Reply:
Yeah, I think we should be able to update `RandomStream` to handle these cases without too much difficulty and/or compromise.